basetest.pm

```diff
@@ -367,6 +368,27 @@ sub next_resultname {
     }
 }
 
+=head2 record_resultfile
+
+    $self->record_resultfile($title, $output [, result => $result] [, resultname => $name]);
+
+Record result file to be parsed when evaluating test results, for example
+within the openQA web interface.
+=cut
+sub record_resultfile {
+    my ($self, $title, $output, %nargs) = @_;
+    my $filename = $self->next_resultname('txt', $nargs{resultname});
```

Comment:
Yes, I suppose evaluating to undef is not an error, but using it with an operator other than `defined` is. I am not sure if my unit tests will work 100% correctly with `Test::Warnings`, so you can assign me a progress ticket to look into it, assuming that is what you are talking about?